src/capi/azext_capi/custom.py

```diff
@@ -271,12 +284,14 @@ def create_workload_cluster(  # pylint: disable=unused-argument,too-many-argumen
         elif location != rg.location:
             msg = "--location is {}, but the resource group {} already exists in {}."
             raise InvalidArgumentValueError(msg.format(location, resource_group_name, rg.location))
-    except CloudError as err:
+    except (CloudError, ResourceNotFoundException) as err:
         if 'could not be found' not in err.message:
             raise
         if not location:
             msg = "--location is required to create the resource group {}."
             raise RequiredArgumentMissingError(msg.format(resource_group_name)) from err
+        if not create_resource_group(cmd, resource_group_name, location, yes):
```

Comment:
as discussed offline we don't want to create the resource group as CAPZ will create it and own its lifecycle if it doesn't already exist. Consider changing the flow to:
- if a resource group exists, confirm with user that they want to reuse it
- if no resource group exists, inform the user that CAPZ will create a resource group as part of cluster creation